### ISSUE TITLE
[BUG] PMM bot crashed and killed process on AWS

### ISSUE THREAD

**Issue by mifeng** [CoinAlpha/hummingbot#1497]
**Describe the bug**
- Running a PMM bot on Binance on a t2.small AWS instance
- Bot crashed for an unknown reason, and killed the process as well
- I will attempt to replicate with the same settings

**Environment**
- AWS Ubuntu 18.04 t2.small instance w/standard settings
- tmux, 1 session

**Screenshots**
<img width="1440" alt="Screen Shot 2020-03-04 at 12 14 50 AM" src="https://user-images.githubusercontent.com/1795030/75860051-0784ea80-5db0-11ea-8649-e15981b9897c.png">

**Release version**
`dev-0.25.0`

**Attachments**
[conf.txt](https://github.com/CoinAlpha/hummingbot/files/4286160/conf.txt)
[log.log](https://github.com/CoinAlpha/hummingbot/files/4286162/log.log)

*WARNING: Do NOT publish any exchange API keys or your wallet's private key. Whoever has access to them may steal your assets!*

**Optional: your discord username:**
**Optional: your ETH address:** (If you want to participate in the bug bounty)

**Comment by dennisocana** [comment]
@Nullably @mifeng based on testing and personal experience this usually happens when HB keeps printing errors. It will flood the logs when it's trying to do something but unsuccessful due to any errors then eventually crashing the instance.

**Comment by mifeng** [comment]
My RLC-ETH bot also crashed:
<img width="1414" alt="Screen Shot 2020-03-04 at 5 53 29 PM" src="https://user-images.githubusercontent.com/1795030/75940349-ac9bd380-5e41-11ea-838e-78ef5ffdc758.png">

[conf.txt](https://github.com/CoinAlpha/hummingbot/files/4290545/conf.txt)
[log.log.zip](https://github.com/CoinAlpha/hummingbot/files/4290546/log.log.zip)

**Comment by mifeng** [comment]
RLC-BTC bot
<img width="1414" alt="Screen Shot 2020-03-04 at 5 53 29 PM" src="https://user-images.githubusercontent.com/1795030/76042626-2e573400-5f0a-11ea-831c-8ed648c39f40.png">

[conf.txt](https://github.com/CoinAlpha/hummingbot/files/4296068/conf.txt)

[log.log 2.zip](https://github.com/CoinAlpha/hummingbot/files/4296070/log.log.2.zip)

**Comment by rennel-tabing** [comment]
I was running an AWS instance for 48 hrs  since Saturday March 7 and did not notice any high utilization using the following config
```
########################################################
###       Pure market making strategy config         ###
########################################################

template_version: 8

############  Basic configuration section  ############

# Exchange and token parameters.
maker_market: binance
maker_market_trading_pair: ONE-USDT

# Size of your bid and ask order.
order_amount: 2500.0

# How far away from mid price to place the bid order.
# Expressed in decimals : 0.01 = 1% away from mid price at that time.
# Example if mid price is 100 and bid_place threshold is 0.01.
# Your bid is placed at 99.
bid_place_threshold: 0.001

# How far away from mid price to place the ask order.
# Expressed in decimals : 0.01 = 1% away from mid price at that time.
# Example if mid price is 100 and ask_place threshold is 0.01.
# Your bid is placed at 101.
ask_place_threshold: 0.001


# Time in seconds before cancelling and rerunning strategy.
# If tick size is 60, the bot cancels active orders and reruns after a minute.
cancel_order_wait_time: 60.0

# Whether to enable advanced configuration setting (true/false).
# Only used when creating a new configuration file.
advanced_mode: true

############  Advanced configuration section  ############

# Ox ONLY - Sets the expiration of limit orders. Minimum is 130 seconds.
expiration_seconds: 130.0

# Whether to single or multiple bids and asks.
mode: multiple

# Number of orders to place on each side of the order book (for multiple orders mode).
number_of_orders: 3

# Size of the first bid and ask order (for multiple orders mode).
order_start_size: 2500.0

# Increment size of consecutive orders after the first order (for multiple orders mode).
order_step_size: 0.0

# Order price space between orders (for multiple orders mode).
order_interval_percent: 0.01

# Whether to enable Inventory skew feature (true/false).
inventory_skew_enabled: false

# Target base asset inventory percentage target to be maintained (for Inventory skew feature).
inventory_target_base_percent: 0.5

# How long to wait before placing the next order in case your order gets filled (for single order mode).
filled_order_replenish_wait_time: 1.0

# Whether to stop cancellations of orders on the other side when one side is filled (true/false).
enable_order_filled_stop_cancellation: false

# Whether to enable best bid ask jumping mode (true/false).
best_bid_ask_jump_mode: false

# The depth in base asset amount to be used for finding top bid and ask (for best bid ask jumping mode).
best_bid_ask_jump_orders_depth: 0.0

# Whether to enable adding transaction costs to order price calculation (true/false).
add_transaction_costs: true

# Whether to use external pricing source for the mid price (true/false).
external_pricing_source: false

# The type of external pricing source (exchange/feed/custom_api).
external_price_source_type:

# An external exchange name (for external exchange pricing source).
external_price_source_exchange:

# A trading pair for the external exchange (for external exchange pricing source).
external_price_source_exchange_trading_pair:

# A base asset, e.g. ETH (for external feed pricing source).
external_price_source_feed_base_asset:

# A quote asset, e.g. USD (for external feed pricing source).
external_price_source_feed_quote_asset:

# An external api that returns price (for external custom_api pricing source).
external_price_source_custom_api:

# For more detailed information, see:
# https://docs.hummingbot.io/strategies/pure-market-making/#configuration-parameters

```

version 0.24.1 - from source

**Comment by gabririgo** [comment]
curious as to whether this issue resulted in the AWS instance to crash or get stuck on high CPU usage?

**Comment by mifeng** [comment]
@gabririgo Yes, we think we've identified the underlying issue that causes this bug as well as AWS instance crashes due to high CPU usage: https://github.com/CoinAlpha/hummingbot/pull/1503

We're testing this fix and if it works, we will deploy to the `development` branch this week.

**Comment by gabririgo** [comment]
> @gabririgo Yes, we think we've identified the underlying issue that causes this bug as well as AWS instance crashes due to high CPU usage: #1503
> 
> We're testing this fix and if it works, we will deploy to the `development` branch this week.

well done!

**Comment by rennel-tabing** [comment]
User Jin reported a similar issue

Release:
latest version 0.25.0 (as of this writing)
He is running 2 hummingbot instance on his AWS server.

Log file:
[jin-logs_conf_pm.log](https://github.com/CoinAlpha/hummingbot/files/4444294/jin-logs_conf_pm.log)
Strategy file:
[Jin-strategy.txt](https://github.com/CoinAlpha/hummingbot/files/4444298/Jin-strategy.txt)

User confirms that he did several parameter changes resulting to stopping and starting the bot.
I advised that he do a fresh strategy configuration start instead.
After 9 hours of implementing a newly started hummingbot instance, his bot becomes unresponsive again and shows a gradual increase of the CPU utilization followed by a huge spike
![image](https://user-images.githubusercontent.com/53868268/78670758-4117ac80-78ac-11ea-922e-5331defea7ec.png)



====================================================
On my test environment  I used the following config
[conf_PMM.txt](https://github.com/CoinAlpha/hummingbot/files/4444429/conf_PMM.txt)

My test environment used the following release version 
0.25.0
dev-0.26.0
```
"branch": "development",
                "commit": "3c620c0a49bab611c69184b109351db26a9c0893",
                "date": "2020-04-04T05:11:05Z",
```

[logs_PMM-dev_0.26.0.log](https://github.com/CoinAlpha/hummingbot/files/4444444/logs_PMM-dev_0.26.0.log)
Steps to reproduce
1. Start a Pure Market making strategy
2. Do a multiple/consecutive stop and start command inside the hummingbot CLI

Instance utilization using 0.25.0
![image](https://user-images.githubusercontent.com/53868268/78672718-08c59d80-78af-11ea-860d-6c00758e2571.png)

Instance utilization using dev-0.26.0
![image](https://user-images.githubusercontent.com/53868268/78672803-21ce4e80-78af-11ea-860c-9ea750cd715c.png)

**Comment by mifeng** [comment]
Fixed by #1683
